### PR TITLE
chore(IDX): replace PAT with GitHub App

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

Following PR #3689 , we replace PATs generated by service accounts with GitHub Apps.

### CODE REVIEW

File: .github/workflows/update-snapshots.yml [hunk 1]
```diff
@@ -31,7 +39,7 @@ jobs:
         with:
           repository: ${{ github.event.pull_request.head.repo.full_name }}
           ref: ${{ github.event.pull_request.head.ref }}
-          token: ${{ secrets.GIX_CREATE_PR_PAT }}
+          token: ${{ steps.app-token.outputs.token }}
```
Comment (cgundy) [inline .github/workflows/update-snapshots.yml:42]: So I'm wondering if this is needed at all here. It looks like the `oisy-backend` action only pushes a docker image, it doesn't create a PR, so I think you could remove the token here and add permissions to the workflow.

File: .github/workflows/update-snapshots.yml [hunk 2]
```diff
@@ -21,6 +21,14 @@ jobs:
         os: [ubuntu-24.04, macos-14]
     if: ${{ github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'run-e2e-snapshots') }}
```
Comment (cgundy) [inline .github/workflows/update-snapshots.yml:22]: Here you would add:
```
permissions:
    contents: write
    packages: write
```

**Review by cgundy — APPROVED** [review]

File: .github/workflows/update-snapshots.yml (hunk 1 above)
Comment (AntonioVentilii) [reply .github/workflows/update-snapshots.yml:42]: You are right! Removed

File: .github/workflows/update-snapshots.yml (hunk 2 above)
Comment (AntonioVentilii) [reply .github/workflows/update-snapshots.yml:22]: included, tks!

**Review by peterpeterparker — APPROVED** [review]
Rubber stamp.

**Pull request merged**